src/components/text/index.tsx

```diff
@@ -62,47 +65,90 @@ class Text extends PureComponent<PropsTypes> {
   //   this._root.setNativeProps(nativeProps); // eslint-disable-line
   // }
 
-  getTextPartsByHighlight(targetString = '', highlightString = '') {
-    if (_.isEmpty(highlightString.trim())) {
-      return [targetString];
+  getPartsByHighlight(targetString = '', highlightString: string | string[]) {
+    if (typeof highlightString === 'string') {
+      if (_.isEmpty(highlightString.trim())) {
+        return [{string: targetString, shouldHighlight: false}];
+      }
+      return this.getTextPartsByHighlight(targetString, highlightString);
+    } else {
+      return this.getArrayPartsByHighlight(targetString, highlightString);
     }
+  }
 
+  getTextPartsByHighlight(targetString = '', highlightString = '') {
+    if (highlightString === '') {
+      return [{string: targetString, shouldHighlight: false}];
+    }
     const textParts = [];
     let highlightIndex;
-
     do {
       highlightIndex = targetString.toLowerCase().indexOf(highlightString.toLowerCase());
       if (highlightIndex !== -1) {
         if (highlightIndex > 0) {
-          textParts.push(targetString.substring(0, highlightIndex));
+          textParts.push({string: targetString.substring(0, highlightIndex), shouldHighlight: false});
         }
-        textParts.push(targetString.substr(highlightIndex, highlightString.length));
+        textParts.push({string: targetString.substr(highlightIndex, highlightString.length), shouldHighlight: true});
         targetString = targetString.substr(highlightIndex + highlightString.length);
       } else {
-        textParts.push(targetString);
+        textParts.push({string: targetString, shouldHighlight: false});
       }
     } while (highlightIndex !== -1);
 
     return textParts;
   }
 
+  getArrayPartsByHighlight(targetString = '', highlightString = ['']) {
+    const target = _.toLower(targetString);
+    const indices = [];
+    let index = 0;
+    let lastWordLength = 0;
+    for (let j = 0; j < highlightString.length; j++) {
+      const word = _.toLower(highlightString[j]);
+      const targetSuffix = target.substring(index + lastWordLength);
+      const i = targetSuffix.indexOf(word);
+      if (i >= 0) {
+        const newIndex = index + lastWordLength + i;
+        indices.push(index + lastWordLength + i);
+        indices.push(index + lastWordLength + i + word.length);
+        index = newIndex;
+        lastWordLength = word.length;
+      } else {
+        break;
+      }
+    }
+    const parts = [];
+    let val = false;
+    for (let k = 0; k < indices.length; k++) {
```

Comment:
I think I can't use map or forEach here since in every iteration I use the current index and the next one (k and k+1) which is not possible with those methods..

Reply:
Yes, map and forEach that comes with javascript don't retrieve the Index..
But using lodash's `_.forEach` or `_.map`, you also have access to the index

```
_.map(array, (item, index) => console.log(index))
_.forEach(array, (item, index) => console.log(index))
```

Reply:
I see, but I need the `item` and the next `item`.. 
I can use it that way:
`_.forEach(array, (item, index) => console.log(item, array[index + 1]))
`
but I don't think it makes that more clear..

Reply:
Ok, I have one last suggestion, let me know what you think (: 

What if instead of keeping the indices as couples (which I personally find confusing) you'll keep them as objects of {start, end}, so indices looks like this
```
[{start: 2, end: 6}, {start: 10, end: 12}, ...]
```
I think it'll make a easier to understand (: 
WDYT?

Reply:
Cool, done :)